transport/rif_comms/client.py

```diff
@@ -92,28 +89,15 @@ def unsubscribe_from(self, rsk_address: Address):
         Invokes the CloseTopic GRPC API endpoint.
         :param rsk_address: RSK address which corresponds to the topic which the client is unsubscribing from.
         """
-        topic_id = self._get_peer_id(to_checksum_address(rsk_address))
-        self.stub.CloseTopic(Channel(channelId=topic_id))
-
-    def _get_peer_id(self, rsk_address: Address) -> str:
-        """
-        Gets the peer ID associated with a node RSK address.
-        :param rsk_address: the RSK address which corresponds to the node to locate
-        :return: a string that represents the peer ID that matches the given address
-        :raises:
-            - No peers from routing table:
-                exception: _InactiveRpcError
-                status = StatusCode.UNKNOWN
-                details = "Failed to lookup key! No peers from routing table!"
-        """
-        return self.stub.LocatePeerId(RskAddress(address=to_checksum_address(rsk_address))).address
+        self.stub.CloseTopicWithRskAddress(
+            RskAddress(address=to_checksum_address(rsk_address))
+        )
 
     def disconnect(self):
         """
          Invokes the EndCommunication GRPC API endpoint.
          Disconnects from RIF Communications Node. Closes grpc connection
         """
-        self.stub.EndCommunication(Void())
+        # FIXME: ending communications should not need variable assignment
```

Comment:
this helps a lot in reducing the flakiness of the tear-down process.

i have no idea why, but we have issues to look into this (see PR description)